cloud/azure/azure_rm_networkinterface.py

```diff
@@ -0,0 +1,587 @@
+
+    if nic.ip_configurations[0].subnet:
+        result['ip_configuration']['subnet']['id'] = \
+            nic.ip_configurations[0].subnet.id
+        id_keys = azure_id_to_dict(nic.ip_configurations[0].subnet.id)
+        result['ip_configuration']['subnet']['virtual_network_name'] = id_keys['virtualNetworks']
+        result['ip_configuration']['subnet']['name'] = id_keys['subnets']
+
+    if nic.ip_configurations[0].public_ip_address:
+        result['ip_configuration']['public_ip_address']['id'] = \
+            nic.ip_configurations[0].public_ip_address.id
+        id_keys = azure_id_to_dict(nic.ip_configurations[0].public_ip_address.id)
+        result['ip_configuration']['public_ip_address']['name'] = id_keys['publicIPAddresses']
+
+    return result
+
+
+class AzureRMNetworkInterface(AzureRMModuleBase):
+
+    def __init__(self, **kwargs):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(required=True),
+            name=dict(required=True),
+            location=dict(type='str'),
+            security_group_name=dict(type='str', aliases=['security_group']),
+            state=dict(default='present', choices=['present', 'absent']),
+            private_ip_address=dict(type='str'),
+            private_ip_allocation_method=dict(type='str', choices=['Dynamic', 'Static'], default='Dynamic'),
+            public_ip_address_name=dict(type='str', aliases=['public_ip_address', 'public_ip_name']),
+            public_ip=dict(type='bool', default=True),
+            subnet_name=dict(type='str', aliases=['subnet']),
+            virtual_network_name=dict(type='str', aliases=['virtual_network']),
+            os_type=dict(type='str', choices=['Windows', 'Linux'], default='Linux'),
+            open_ports=dict(type='list'),
+            public_ip_allocation_method=dict(type='str', choices=['Dynamic', 'Static'], default='Dynamic'),
+        )
+
+        super(AzureRMNetworkInterface, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                      supports_check_mode=True, **kwargs)
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.security_group_name = None
+        self.private_ip_address = None
+        self.private_ip_allocation_method = None
+        self.public_ip_address_name = None
+        self.state = None
+        self.subnet_name = None
+        self.tags = None
+        self.virtual_network_name = None
+        self.security_group_name = None
+        self.os_type = None
+        self.open_ports = None
+        self.public_ip_allocation_method = None
+        self.public_ip = None
+
+        self.results = dict(
+            changed=False,
+            check_mode=self.check_mode,
+            results=dict()
+        )
+
+    def exec_module_impl(self, **kwargs):
+
+        for key in self.module_arg_spec.keys() + ['tags']:
+            setattr(self, key, kwargs[key])
```

Comment:
Instead of passing the args to this function, can just use them directly from self.module.params:
``` python
setattr(self, key, self.module.params[key])
```

This also probably belongs in ```__init__```.  It just has to be done after the call to the base class's ``__init__``` so that the base class has a chance to create self.module.